src/lib/edge-functions/proxy.mjs

```diff
@@ -167,28 +167,22 @@ export const initializeProxy = async ({
     }
 
     const featureFlags = ['edge_functions_bootstrap_failure_mode']
-    const forwardedHost = `localhost:${passthroughPort}`
 
     req[headersSymbol] = {
       [headers.FeatureFlags]: getFeatureFlagsHeader(featureFlags),
-      [headers.ForwardedHost]: forwardedHost,
+      [headers.ForwardedProtocol]: req.socket.encrypted ? 'https:' : 'http:',
```

Comment:
Can we make [the caller method](https://github.com/netlify/cli/blob/58b40b3af29a015434b99d854a841f146e291125/src/utils/proxy.mjs#L666) pass `settings.https` rather than using `req.socket.encrypted` here? That way we have a unified detection mechanism for whether we're running on HTTPS.

Reply:
done in dfa73c9881fb44405d6fcf04e8ea738b0bf1c451